Packs/AzureDevOps/Integrations/AzureDevOps/AzureDevOps.py

```diff
@@ -0,0 +1,1732 @@
+            limit (int): The number of results to retrieve.
+            continuation_token (str): A continuation token from a previous request, to retrieve the next page of results.
+
+        Returns:
+            Response: API response from Azure.
+
+        """
+        params = remove_empty_elements({'api-version': '6.1-preview.1',
+                                        '$top': limit,
+                                        'continuationToken': continuation_token,
+                                        'filter': 'heads'})
+
+        url_suffix = f'{project}/_apis/git/repositories/{repository}/refs'
+
+        response = self.ms_client.http_request(method='GET',
+                                               url_suffix=url_suffix,
+                                               params=params,
+                                               resp_type='response')
+
+        return response
+
+
+def generate_pipeline_run_output(response: dict, project: str) -> dict:
+    """
+    Create XSOAR context output for retrieving pipeline run information.
+    Args:
+        response (dict): API response from Azure.
+        project (str): The name of the pipeline project.
+
+    Returns:
+        dict: XSOAR command outputs.
+
+    """
+    str_time = response.get('createdDate')
+    str_time = str_time[:-2] + 'Z'
+    creation_date = FormatIso8601(datetime.strptime(str_time, DATE_FORMAT))
+
+    run_data = {"state": response.get('state'),
+                "createdDate": creation_date,
+                "url": response.get('url'),
+                "id": response.get('id'),
+                "name": response.get('name'),
+                "result": response.get('result', 'unknown')
+                }
+
+    pipeline_data = {"id": dict_safe_get(response, ['pipeline', 'id']),
+                     "name": dict_safe_get(response, ['pipeline', 'name']),
+                     "revision": dict_safe_get(response, ['pipeline', 'revision']),
+                     "folder": dict_safe_get(response, ['pipeline', 'folder']),
+                     "Run": run_data
+                     }
+
+    outputs = {"name": project, "Pipeline": pipeline_data}
+
+    return outputs
+
+
+def filter_pipeline_run_table(run: dict) -> dict:
+    """
+    Filter pipeline-run required information for representing to the user.
+    Args:
+        run (dict): Pipeline-run information.
+
+    Returns:
+        dict: Filtered pipeline-run information.
+
+    """
+    str_time = run.get('createdDate')
+    str_time = str_time[:-2] + 'Z'
+    creation_date = FormatIso8601(datetime.strptime(str_time, DATE_FORMAT))
+
+    return {
+        "pipeline_id": dict_safe_get(run, ['pipeline', 'id']),
+        "run_state": run.get('state'),
+        "creation_date": creation_date,
+        "run_id": run.get('id'),
+        "result": run.get('result', 'unknown')
+    }
+
+
+def generate_pipeline_run_readable_information(response: dict, is_multiple_response: bool = False,
```

Comment:
I understand the logic behind it and I even thought of implementing it that way.
However, I tried to separate these two objects - I tried to think of cases where we'll need to change the structure of the output - this will lead to further changes in the readable output as well.